sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -35,6 +35,9 @@ object UnsupportedOperationChecker {
       case p if p.isStreaming =>
         throwError("Queries with streaming sources must be executed with writeStream.start()")(p)
 
+      case p: Deduplication =>
+        throwError("Batch queries should not use Deduplication")(p)
```

Comment:
because of this, I would prefer the naming to imply that as well. Maybe rename `Deduplication` to `StreamingDeduplication` or something.